experimental/Schemes/Schemes.jl

```diff
@@ -0,0 +1,2587 @@
+    # TODO: Implement cheap plausibility checks.
+    return new( parent, covering, transitions )
+  end
+end
+
+@Markdown.doc """
+    function LineBundle( L::LineBundle )
+
+Returns a copy of L.
+"""
+function LineBundle( L::LineBundle )
+  return LineBundle( L.parent, L.covering, copy(L.transitions) )
+end
+
+function Base.copy( L::LineBundle )
+  return LineBundle( L )
+end
+
+@Markdown.doc """
+    parent( L::LineBundle ) -> CoveredScheme
+
+Returns the CoveredScheme X on which L is defined.
+"""
+parent( L::LineBundle ) = L.parent
+@Markdown.doc """
+    covering( L::LineBundle ) -> Covering
+
+Returns the Covering C of the CoveredScheme X on which L is defined.
+"""
+covering( L::LineBundle ) = L.covering
+@Markdown.doc """
+    transitions( L::LineBundle ) -> Matrix{MPolyElem}
+
+Returns the matrix of transition functions for L for 
+pairs of affine patches in the Covering C of the 
+CoveredScheme X on which L is defined. 
+"""
+transitions( L::LineBundle ) = L.transitions
+
+@Markdown.doc """
+    function OO( k::Ring, n::Int, d::Int ) -> LineBundle
+
+Returns Serre's twisting sheaf 𝒪(d) on ℙₖⁿ, trivialized in 
+the canonical covering.
+"""
+function OO( k::Ring, n::Int, d::Int )
+  IPn = projective_space( k, n )
+  C = coverings( IPn )[1]
+  transitions = Array{MPolyElem,2}(undef,n+1,n+1)
+  if d >= 0
+    for i in OrderedMultiindex(n+1,2)
+      a = inverted_element(overlap(glueings(C)[getindex(i,1),getindex(i,2)]))
+      transitions[getindex(i,1), getindex(i,2)] = a^d
+      #transitions[getindex(i,1), getindex(i,2)] = (inverted_element(overlap(glueings[i])))^d
+      transitions[i[2],i[1]] = (denom(overlap(glueings(C)[i[1],i[2]])))^d
+    end
+  else
+    d = -d
+    for i in OrderedMultiindex(n+1,2)
+      a = inverted_element(overlap(glueings(C)[getindex(i,1),getindex(i,2)]))
+      transitions[getindex(i,2), getindex(i,1)] = a^d
+      #transitions[getindex(i,1), getindex(i,2)] = (inverted_element(overlap(glueings(C)[i])))^d
+      transitions[i[1],i[2]] = (denom(overlap(glueings(C)[getindex(i,1),getindex(i,2)])))^d
+    end
+  end
+  L = LineBundle( IPn, C, transitions )
+  L.name = "OO($n,$d)"
+  return L
+end
+
+@Markdown.doc """
+    OO( n::Int, d::Int ) = OO( QQ, n, d )
+"""
+OO( n::Int, d::Int ) = OO( QQ, n, d )
+
+@Markdown.doc """
+    function Base.show( io::Base.IO, L::LineBundle )
+
+Print the LineBundle L to the stream io. Whenever the 
+field L.name is set, this output is abbreviated to printing 
+that name. 
+"""
+function Base.show( io::Base.IO, L::LineBundle )
+  if isdefined( L, :name )
+    Base.print( io, L.name )
+    return
+  end
+  Base.print( io, "Line bundle on " )
+  Base.print( io, L.parent )
+  #Base.print( io, " with respect to the covering " )
+  #Base.print( io, L.covering )
+  Base.print( io, " given by the transition functions\n" )
+  n = length( patches(L.covering) )
+  for i in (1:n)
+    for j in (1:n)
+      if i == j 
+	Base.println( io, "($i,$j): 1" )
```

Comment:
Broken indentation (or perhaps tabs mixed with spaces? amounts to the same to me, though...)


Perhaps most efficient would be if you run something like `JuliaFormatter` once on this file